internal/compiler/transforms/compile/css-handler/prefix/utils.ts

```diff
@@ -0,0 +1,246 @@
+		);
+	}
+	return false;
+}
+
+export function nodeHasPrefixedPropertyValue(
+	node: CSSBlock,
+	property: string,
+	value: string,
+	prefix: string,
+	rename: (value: string) => string = (value) => value,
+): boolean {
+	if (node.value) {
+		return node.value.some((n) =>
+			n.type === "CSSDeclaration" &&
+			n.name === property &&
+			n.value.length === 1 &&
+			n.value[0]?.type === "CSSIdentifier" &&
+			n.value[0].value === rename(`-${prefix}-${value}`)
+		);
+	}
+	return false;
+}
+
+export function nodePropertyIndex(node: CSSBlock, property: string): number {
+	return node.value
+		? node.value.findIndex((n) =>
+				n.type === "CSSDeclaration" && n.name === property
+			)
+		: -1;
+}
+
+export function nodePropertyValueIndex(
+	node: CSSBlock,
+	property: string,
+	value: string,
+): number {
+	return node.value
+		? node.value.findIndex((n) =>
+				n.type === "CSSDeclaration" &&
+				n.name === property &&
+				n.value.length === 1 &&
+				n.value[0]?.type === "CSSIdentifier" &&
+				n.value[0].value === value
+			)
+		: -1;
+}
+export interface PrefixVisitor<State extends UnknownObject> {
+	name: string;
+	enter?: (
+		path: CompilerPath,
+		targets: Browser[],
+		state: VisitorStateEnter<State>,
+	) => EnterSignal;
+	exit?: (
+		path: CompilerPath,
+		targets: Browser[],
+		state: VisitorStateExit<State>,
+	) => ExitSignal;
+}
+
+export function createPrefixVisitor<State extends UnknownObject>(
+	visitor: PrefixVisitor<State>,
+) {
+	return {
+		name: `css-handler/prefix/${visitor.name}`,
+		enter: visitor.enter,
+		exit: visitor.exit,
+	};
+}
+
+export function wrapPrefixVisitor<State extends UnknownObject>(
+	visitor: PrefixVisitor<State>,
+	targets: Browser[],
+) {
+	return createVisitor<State>({
+		name: visitor.name,
+		enter: visitor.enter != null
+			? (path, state) => visitor.enter!(path, targets, state)
+			: undefined,
+		exit: visitor.exit != null
+			? (path, state) => visitor.exit!(path, targets, state)
+			: undefined,
+	});
+}
+
+interface PrefixCSSPropertyProps {
+	path: CompilerPath;
+	propertyName: string;
+	browserFeaturesKey: string;
+	targets: Browser[];
+	rename?: (propertyName: string) => string;
+}
+
+interface PrefixCSSValueProps extends PrefixCSSPropertyProps {
+	value: string;
+	rename?: (value: string) => string;
+}
+
+export function prefixCSSProperty(
+	{
+		path,
+		propertyName,
+		browserFeaturesKey,
+		targets,
+		rename = (propertyName) => propertyName,
+	}: PrefixCSSPropertyProps,
+) {
+	const {node} = path;
+	if (node.type === "CSSBlock") {
+		if (node.value && node.value.length > 0) {
+			const propertyIndex = nodePropertyIndex(node, propertyName);
+			if (propertyIndex > -1) {
+				const property = node.value[propertyIndex] as CSSDeclaration;
+				const newDeclarations = [];
+				const prefixes = new Set(
+					targets.filter((browser) =>
+						browser.cssFeatureRequiresPrefix(browserFeaturesKey)
+					).map((browser) => browser.getPrefix()),
+				);
```

Comment:
Could this be simplified as:

```suggestion
				const prefixes = new Set();

				for (const browser of targets) {
					if (browser.cssFeatureRequiresPrefix(browserFeaturesKey)) {
						prefixes.add(browser.getPrefix());
					}
				}
```

Before we were doing two loop iterations, creating functions, and had more indirection. We should optimize and consider the case where we have literally megabytes of CSS in a file, and these sorts of operations add up.